packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -609,23 +602,24 @@ function initializeMode(vm: VM) {
         );
     }
 
+    // Default modes.
+    vm.renderMode = RenderMode.Shadow;
+    vm.shadowMode = ShadowMode.Native;
+
     if (renderMode === 'light') {
         vm.renderMode = RenderMode.Light;
-        vm.shadowMode = ShadowMode.None;
-        return;
+    } else if (renderer.ssr) {
+        vm.renderMode = RenderMode.Shadow;
     }
 
     if (renderer.ssr) {
-        vm.renderMode = RenderMode.Shadow;
         vm.shadowMode = ShadowMode.Native;
-        return;
+    } else if (renderer.syntheticShadow) {
+        vm.shadowMode =
+            isTrue(preferNativeShadow) && isNativeShadowRootDefined
+                ? ShadowMode.Native
```

Comment:
Being able to set native mode in the context of the synthetic shadow polyfill was the ultimate goal of all these changes. We were making a lot of decisions based on whether the polyfill was applied or not, when that's not how mixed mode operates.